idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/action/AnActionSyntax.kt

```diff
@@ -10,55 +10,113 @@ import com.intellij.openapi.actionSystem.TimerListener
 import com.intellij.openapi.application.ModalityState
 import javax.swing.Icon
 
-// TODO: Check Default of actionId
+/**
+ * [AnActionExtensionProvider] is in charge of the lifecycle of [AnAction].
+ * [AnAction] is an overly broad concept to access and execute nearly anything in the editor.
```

Comment:
That sounds a bit critical on the concept of `AnAction` ;) I agree that it's a broad concept.

imho, the main point of an action is that it provides the functionality to be shown in menus, toolbars, context menus and more. It's sometimes used to just contain an computation on-demand without the ui part, but it's usually connected with a piece of the ui. An action can be reused - a single instance can be added to several places and it's always updated properly (via method `AnAction.update()` and the presentation attached to the event).
Of course, the implementations of an action need to be able to run arbitrary code.

Reply:
What do you suggest we should use instead?
I didn't mean to sound critical, though both of us agree on the essence if I am not mistaken. Does this sound better?
```
[AnAction] can be placed, executed and composed into any workflow, 
throughout the whole application lifecycle, 
whether on-demand or as a standalone workflow. 
They're able to access almost every part of the ide.
```
I'd appreciate @jkmcclellan feedback on this 😄

Reply:
Sounds pretty straightforward to me! Your modification does make it clearer.

Reply:
Sounds good to me

Reply:
Sounds good. Then I'll swap this out.